# PINIR Data Analysis Page-1

In [1]:
import pandas as pd
import numpy as np

In [2]:
pd.set_option('display.max_columns',None)
pd.set_option('display.max_rows',None)

In [3]:
pd.reset_option('display.max_columns')
pd.reset_option('display.max_rows')

In [4]:
pd.set_option('display.max_colwidth',2000)

In [5]:
pd.reset_option('display.max_colwidth')

## Organism Wise Distribution of PIN II PIs

In [6]:
seq=pd.read_csv('data/24_Sequence_Organism_IsoelectricPoint_Amino_Composition.csv')

In [7]:
seq.columns

Index(['UniprotID', 'ProteinNames', 'GeneNames', 'OrganismID', 'Organism',
       'Genus', 'Species', 'Solanaceae', 'Function[CC]', 'Sequence', 'Length',
       'Mass', 'average_pI', 'pH_5.5_charge', 'pH_7.4_charge', 'pH_8.0_charge',
       'A', 'A_per', 'C', 'C_per', 'D', 'D_per', 'E', 'E_per', 'F', 'F_per',
       'G', 'G_per', 'H', 'H_per', 'I', 'I_per', 'K', 'K_per', 'L', 'L_per',
       'M', 'M_per', 'N', 'N_per', 'O', 'O_per', 'P', 'P_per', 'Q', 'Q_per',
       'R', 'R_per', 'S', 'S_per', 'T', 'T_per', 'U', 'U_per', 'V', 'V_per',
       'W', 'W_per', 'X', 'X_per', 'Y', 'Y_per', 'Z', 'Z_per'],
      dtype='object')

In [9]:
seq_org=seq[['UniprotID','OrganismID', 'Organism',
       'Genus', 'Species', 'Solanaceae', 'Function[CC]', 'Length',
       'Mass']]

In [11]:
seq_org.head(2)

UniprotID  OrganismID                                           Organism  \
0    P05119        4081  Solanum lycopersicum (Tomato) (Lycopersicon es...   
1    P84813        4113                         Solanum tuberosum (Potato)   

     Genus                Species  Solanaceae  \
0  Solanum  Solanum lycopersicum            1   
1  Solanum     Solanum tuberosum            1   

                                        Function[CC]  Length     Mass  
0  FUNCTION: Potent inhibitor of both trypsin and...     148  16293.0  
1  FUNCTION: Inhibitor of serine proteases chymot...      47   4833.0

In [13]:
seq_org.OrganismID.nunique()## Distributed across 125 Organisms

125

In [14]:
## Organism wise Distribution of PIN II PI sequences

In [15]:
seq_org.Organism.value_counts()

Capsicum annuum (Capsicum pepper)                          99
Solanum tuberosum (Potato)                                 52
Nicotiana tabacum (Common tobacco)                         22
Solanum lycopersicum (Tomato) (Lycopersicon esculentum)    21
Capsicum chinense (Scotch bonnet) (Bonnet pepper)          13
                                                           ..
Oryza punctata (Red rice)                                   1
Lupinus angustifolius (Narrow-leaved blue lupine)           1
Punica granatum (Pomegranate)                               1
Nicotiana miersii                                           1
Oryza sativa subsp. indica (Rice)                           1
Name: Organism, Length: 125, dtype: int64

In [16]:
## PIN II PI sequences distribution in Solanaceae and non Solanaceae plants

In [17]:
seq_org.Solanaceae.value_counts()

1    288
0    164
Name: Solanaceae, dtype: int64

In [18]:
## PIN II PI sequences distribution in Top 10 Solanaceae plants

In [20]:
filt=seq_org.Solanaceae==1
seq_org.loc[filt,:].Organism.value_counts().head(10)

Capsicum annuum (Capsicum pepper)                               99
Solanum tuberosum (Potato)                                      52
Nicotiana tabacum (Common tobacco)                              22
Solanum lycopersicum (Tomato) (Lycopersicon esculentum)         21
Capsicum chinense (Scotch bonnet) (Bonnet pepper)               13
Nicotiana attenuata (Coyote tobacco)                            11
Capsicum baccatum (Peruvian pepper)                             11
Solanum chilense (Tomato) (Lycopersicon chilense)               11
Nicotiana sylvestris (Wood tobacco) (South American tobacco)    10
Solanum chacoense (Chaco potato)                                 5
Name: Organism, dtype: int64

In [21]:
## PIN II PI sequences distribution in Top 10 non-Solanaceae plants

In [22]:
filt=seq_org.Solanaceae==0
seq_org.loc[filt,:].Organism.value_counts().head(10)

Selaginella moellendorffii (Spikemoss)                              7
Coffea canephora (Robusta coffee)                                   6
Rosa chinensis (China rose)                                         5
Dichanthelium oligosanthes                                          4
Cuscuta australis                                                   4
Arachis hypogaea (Peanut)                                           4
Panicum hallii var. hallii                                          4
Cuscuta campestris                                                  4
Noccaea caerulescens (Alpine penny-cress) (Thlaspi caerulescens)    4
Eragrostis curvula (weeping love grass)                             3
Name: Organism, dtype: int64

##### Above results show: Out of 452 sequences, 288 Pin-II Sequences are from Solanaceae plants and the remaining 164 are from non-Solanaceae plants. Distribution of Pin-II sequences in top 3 Species is as follows: Capsicum annuum 99;Solanum tuberosum 52; and Nicotiana tabacum 22

## Domain (IRDs) Wise Distribution of PIN II PIs

In [23]:
pinir=pd.read_csv('data/19_Sequences_Domains_RCL_Linker_Organisms.csv')

In [24]:
pinir.head(2)

UniprotID                                           Sequence  OrganismID  \
0    P05119  MAVHKEVNFVAYLLIVLGMFLYVDAKACTRECGNLGFGICPRSEGS...        4081   
1    P05119  MAVHKEVNFVAYLLIVLGMFLYVDAKACTRECGNLGFGICPRSEGS...        4081   

                                            Organism    Genus  \
0  Solanum lycopersicum (Tomato) (Lycopersicon es...  Solanum   
1  Solanum lycopersicum (Tomato) (Lycopersicon es...  Solanum   

                 Species  Solanaceae DomainID  \
0  Solanum lycopersicum            1   IRD-62   
1  Solanum lycopersicum            1  IRD-405   

                                              Domain  DomainStartPosition  \
0  PICINCCSGYKGCNYYNSFGKFICEGESDPKRPNACTFNCDPNIAY...                   49   
1  NACTFNCDPNIAYSRCPRSQGKSLIYPTGCTTCCTGYKGCYYFGKD...                   82   

   ...  P1primeResidue  targetProtease  tpID RCL_startPosition  \
0  ...               N    Chymotrypsin  TP-2              35.0   
1  ...               N    Chymotrypsin  TP-2               2.0   

  RCL_endPosition linkerID Linker LinkerType Linker_startPosition  \
0            39.0     L-21  DPKRP        1.0                 28.0   
1             6.0      NaN    NaN        NaN                  NaN   

   Linker_endPosition  
0                32.0  
1                 NaN  

[2 rows x 26 columns]

In [27]:
pinir.shape

(1309, 26)

In [28]:
pinir.columns

Index(['UniprotID', 'Sequence', 'OrganismID', 'Organism', 'Genus', 'Species',
       'Solanaceae', 'DomainID', 'Domain', 'DomainStartPosition',
       'DomainEndPosition', 'DomainType', 'RclID', 'RCL', 'P2Residue',
       'P1Residue', 'P1primeResidue', 'targetProtease', 'tpID',
       'RCL_startPosition', 'RCL_endPosition', 'linkerID', 'Linker',
       'LinkerType', 'Linker_startPosition', 'Linker_endPosition'],
      dtype='object')

In [29]:
seq_domain=pinir[['UniprotID', 'OrganismID', 'Organism', 'Genus', 'Species',
       'Solanaceae', 'DomainID','DomainType', 'RclID', 'RCL', 'P2Residue',
       'P1Residue', 'P1primeResidue', 'targetProtease', 'tpID',
       'linkerID', 'Linker',
       'LinkerType']]

In [30]:
seq_domain.head(2)

UniprotID  OrganismID                                           Organism  \
0    P05119        4081  Solanum lycopersicum (Tomato) (Lycopersicon es...   
1    P05119        4081  Solanum lycopersicum (Tomato) (Lycopersicon es...   

     Genus                Species  Solanaceae DomainID  DomainType RclID  \
0  Solanum  Solanum lycopersicum            1   IRD-62         1.0  RL-7   
1  Solanum  Solanum lycopersicum            1  IRD-405         3.0  RL-7   

     RCL P2Residue P1Residue P1primeResidue targetProtease  tpID linkerID  \
0  CTFNC         T         F              N   Chymotrypsin  TP-2     L-21   
1  CTFNC         T         F              N   Chymotrypsin  TP-2      NaN   

  Linker  LinkerType  
0  DPKRP         1.0  
1    NaN         NaN

In [31]:
seq_domain.shape

(1309, 18)

In [32]:
seq_domain.UniprotID.nunique()

387

In [33]:
seq_domain.DomainID.nunique()

690

### Domain Frequency In PINIR

In [36]:
## seq_domain.DomainID.value_counts().to_csv('data/25a_DomainFrequencyInPINIR.csv')

In [34]:
seq_domain.DomainID.value_counts()

IRD-133    39
IRD-84     29
IRD-536    25
IRD-559    23
IRD-122    19
           ..
IRD-105     1
IRD-565     1
IRD-332     1
IRD-151     1
IRD-269     1
Name: DomainID, Length: 690, dtype: int64

#### From the above result we can say that IRD-133 is found 39 times in seq_domain table. But we can't say it is found in 39 unique sequences because it might occur more than once in a sequence. To find unique sequences frquency we have to filter out duplicated records

In [38]:
## Sequences in which same Domain is found more than once

In [37]:
seq_domain.duplicated().sum()

59

In [39]:
seq_domain.loc[~seq_domain.duplicated(),:].UniprotID.count()

1250

In [40]:
seq_uniqueDomain=seq_domain.loc[~seq_domain.duplicated(),:]

In [41]:
seq_uniqueDomain.duplicated().sum()

0

In [43]:
seq_uniqueDomain.DomainID.value_counts().to_csv('data/25b_UniqueDomainFrequencyInPINIR.csv')

In [44]:
seq_uniqueDomain.DomainID.value_counts()

IRD-133    34
IRD-84     29
IRD-536    25
IRD-559    21
IRD-122    18
           ..
IRD-473     1
IRD-266     1
IRD-643     1
IRD-100     1
IRD-251     1
Name: DomainID, Length: 690, dtype: int64

#### From the above result we can say that IRD-133 is found in 34 unique sequences.

### Domain Frequency in terms of Species in which they are found

In [45]:
seq_uniqueDomain.head(2)

UniprotID  OrganismID                                           Organism  \
0    P05119        4081  Solanum lycopersicum (Tomato) (Lycopersicon es...   
1    P05119        4081  Solanum lycopersicum (Tomato) (Lycopersicon es...   

     Genus                Species  Solanaceae DomainID  DomainType RclID  \
0  Solanum  Solanum lycopersicum            1   IRD-62         1.0  RL-7   
1  Solanum  Solanum lycopersicum            1  IRD-405         3.0  RL-7   

     RCL P2Residue P1Residue P1primeResidue targetProtease  tpID linkerID  \
0  CTFNC         T         F              N   Chymotrypsin  TP-2     L-21   
1  CTFNC         T         F              N   Chymotrypsin  TP-2      NaN   

  Linker  LinkerType  
0  DPKRP         1.0  
1    NaN         NaN

In [51]:
seq_uniqueDomain[['DomainID','OrganismID']].duplicated().sum()

448

In [52]:
seq_uniqueDomain[['DomainID','Genus']].duplicated().sum()

558

In [53]:
seq_uniqueDomain[['DomainID','Species']].duplicated().sum()

448

In [54]:
orga_domain=seq_uniqueDomain[['DomainID','OrganismID','Organism','Genus','Species','Solanaceae']]

In [55]:
orga_domain.head()

DomainID  OrganismID                                           Organism  \
0   IRD-62        4081  Solanum lycopersicum (Tomato) (Lycopersicon es...   
1  IRD-405        4081  Solanum lycopersicum (Tomato) (Lycopersicon es...   
2  IRD-456        4081  Solanum lycopersicum (Tomato) (Lycopersicon es...   
3   IRD-54        4113                         Solanum tuberosum (Potato)   
4   IRD-56        4113                         Solanum tuberosum (Potato)   

     Genus                Species  Solanaceae  
0  Solanum  Solanum lycopersicum            1  
1  Solanum  Solanum lycopersicum            1  
2  Solanum  Solanum lycopersicum            1  
3  Solanum     Solanum tuberosum            1  
4  Solanum     Solanum tuberosum            1

In [56]:
orga_domain.duplicated().sum()

448

In [57]:
orga_domain[['Species','DomainID']].duplicated().sum()

448

In [58]:
species_uniqueDomains=orga_domain.loc[~orga_domain[['Species','DomainID']].duplicated(),:]

In [59]:
species_uniqueDomains.head()

DomainID  OrganismID                                           Organism  \
0   IRD-62        4081  Solanum lycopersicum (Tomato) (Lycopersicon es...   
1  IRD-405        4081  Solanum lycopersicum (Tomato) (Lycopersicon es...   
2  IRD-456        4081  Solanum lycopersicum (Tomato) (Lycopersicon es...   
3   IRD-54        4113                         Solanum tuberosum (Potato)   
4   IRD-56        4113                         Solanum tuberosum (Potato)   

     Genus                Species  Solanaceae  
0  Solanum  Solanum lycopersicum            1  
1  Solanum  Solanum lycopersicum            1  
2  Solanum  Solanum lycopersicum            1  
3  Solanum     Solanum tuberosum            1  
4  Solanum     Solanum tuberosum            1

In [60]:
species_uniqueDomains.DomainID.value_counts()

IRD-387    7
IRD-176    4
IRD-211    4
IRD-357    3
IRD-360    3
          ..
IRD-555    1
IRD-23     1
IRD-694    1
IRD-543    1
IRD-251    1
Name: DomainID, Length: 690, dtype: int64

#### The above result gives Domain Frequency in terms of species in which they are found

### Distribution of Domains Species wise

In [62]:
species_uniqueDomains.groupby('DomainID').Species.value_counts()

DomainID  Species              
IRD-1     Solanum lycopersicum     1
IRD-10    Capsicum annuum          1
          Capsicum chinense        1
IRD-100   Capsicum annuum          1
IRD-101   Capsicum annuum          1
                                  ..
IRD-95    Capsicum annuum          1
IRD-96    Capsicum annuum          1
IRD-97    Capsicum annuum          1
IRD-98    Capsicum annuum          1
IRD-99    Capsicum annuum          1
Name: Species, Length: 802, dtype: int64

In [64]:
## species_uniqueDomains.groupby('DomainID').Species.value_counts().to_csv('data/26_UniqueDomainFrequency_In_Species.csv')

## Domain Architectures

In [66]:
seq_domain.head(2)

UniprotID  OrganismID                                           Organism  \
0    P05119        4081  Solanum lycopersicum (Tomato) (Lycopersicon es...   
1    P05119        4081  Solanum lycopersicum (Tomato) (Lycopersicon es...   

     Genus                Species  Solanaceae DomainID  DomainType RclID  \
0  Solanum  Solanum lycopersicum            1   IRD-62         1.0  RL-7   
1  Solanum  Solanum lycopersicum            1  IRD-405         3.0  RL-7   

     RCL P2Residue P1Residue P1primeResidue targetProtease  tpID linkerID  \
0  CTFNC         T         F              N   Chymotrypsin  TP-2     L-21   
1  CTFNC         T         F              N   Chymotrypsin  TP-2      NaN   

  Linker  LinkerType  
0  DPKRP         1.0  
1    NaN         NaN

In [68]:
seq_domain.UniprotID.count()

1309

In [69]:
## Domain Count in Sequences

In [70]:
seq_domain.UniprotID.value_counts()

A0A1U7WDA6    21
Q9SDW8        15
Q1WL31        15
Q1WL35        15
A0A1U7Y2F0    13
              ..
A0A446KEB4     1
I7AXR4         1
D8QVL2         1
Q2TU74         1
M1C870         1
Name: UniprotID, Length: 387, dtype: int64

In [71]:
domArch=seq_domain.UniprotID.value_counts()

In [73]:
## domArch.to_csv('data/27_DomainCOuntInSequences.csv')

In [76]:
## domArch.value_counts().to_csv('data/28_DomainArchitectures.csv')

In [74]:
domArch.value_counts()

1     174
5      77
3      51
2      24
11     17
7      16
4      10
9       6
13      4
15      3
10      2
6       1
12      1
21      1
Name: UniprotID, dtype: int64

#### Above result shows that 14 type of Domain Architectures are found in PIN-II sequences in which sequences with single domains are the most i.e. 174 and 162 PIs consisted domains between 2 to 5.

In [77]:
seq.head(1)

UniprotID                                       ProteinNames GeneNames  \
0    P05119  Wound-induced proteinase inhibitor 2 (Wound-in...       NaN   

   OrganismID                                           Organism    Genus  \
0        4081  Solanum lycopersicum (Tomato) (Lycopersicon es...  Solanum   

                 Species  Solanaceae  \
0  Solanum lycopersicum            1   

                                        Function[CC]  \
0  FUNCTION: Potent inhibitor of both trypsin and...   

                                            Sequence  ...  V    V_per  W  \
0  MAVHKEVNFVAYLLIVLGMFLYVDAKACTRECGNLGFGICPRSEGS...  ...  7  4.72973  0   

   W_per  X  X_per   Y     Y_per  Z  Z_per  
0    0.0  0    0.0  11  7.432432  0      0  

[1 rows x 64 columns]

In [78]:
seq.shape

(452, 64)

In [79]:
seq_with_DomainCount=pd.read_csv('data/27_DomainCOuntInSequences.csv')

In [80]:
seq_with_DomainCount.head(1)

UniprotID  DomainCount
0  A0A1U7WDA6           21

In [81]:
seq_with_DomainCount.shape

(387, 2)

In [82]:
## Creating new Sequence data fame with 'No. of Domains init' information

In [83]:
seq_with_DomainCount=pd.merge(seq_with_DomainCount,seq,on='UniprotID',how='left')

In [84]:
seq_with_DomainCount.shape

(387, 65)

In [85]:
seq_with_DomainCount.head(1)

UniprotID  DomainCount                          ProteinNames  \
0  A0A1U7WDA6           21  uncharacterized protein LOC104227417   

      GeneNames  OrganismID  \
0  LOC104227417        4096   

                                            Organism      Genus  \
0  Nicotiana sylvestris (Wood tobacco) (South Ame...  Nicotiana   

                 Species  Solanaceae Function[CC]  ...   V     V_per  W  \
0  Nicotiana sylvestris            1          NaN  ...  20  2.911208  0   

   W_per  X  X_per   Y    Y_per  Z  Z_per  
0    0.0  0    0.0  24  3.49345  0      0  

[1 rows x 65 columns]

### Solanaceae wise distribution of DomainArchitecture

In [86]:
seq_with_DomainCount.groupby('Solanaceae').DomainCount.value_counts()

Solanaceae  DomainCount
0           1              112
            2                4
1           5               77
            1               62
            3               51
            2               20
            11              17
            7               16
            4               10
            9                6
            13               4
            15               3
            10               2
            6                1
            12               1
            21               1
Name: DomainCount, dtype: int64

In [88]:
# seq_with_DomainCount.groupby('Solanaceae').DomainCount.value_counts().to_csv('data/29_Solanaceae_wise_distribution_of_DomainArchitecture.csv')

In [89]:
## Sequences with Single Domain

In [90]:
seq_with_DomainCount.loc[seq_with_DomainCount.DomainCount==1,:].UniprotID.count()

174

In [91]:
seq_with_singleDomain=seq_with_DomainCount.loc[seq_with_DomainCount.DomainCount==1,:]

### Distribution of Single Domain Architecture Sequences, Solanaceae wise

In [92]:
seq_with_singleDomain.Solanaceae.value_counts()

0    112
1     62
Name: Solanaceae, dtype: int64

### Species wise distribution of Single Domain Architecture Sequences, grouped by Solanaceae

In [93]:
seq_with_singleDomain.groupby('Solanaceae').Species.nunique()

Solanaceae
0    70
1    11
Name: Species, dtype: int64

In [94]:
seq_with_singleDomain.groupby('Solanaceae').Species.value_counts()

Solanaceae  Species                   
0           Coffea canephora              5
            Rosa chinensis                5
            Noccaea caerulescens          4
            Panicum hallii var. hallii    4
            Cuscuta australis             3
                                         ..
1           Nicotiana attenuata           3
            Solanum americanum            2
            Solanum chilense              2
            Solanum lycopersicum          2
            Solanum melongena             1
Name: Species, Length: 81, dtype: int64

In [96]:
## seq_with_singleDomain.groupby('Solanaceae').Species.value_counts().to_csv('data/30_Species_wise_distribution_of_singleDomain_sequences.csv')

### Organism wise distribution of Domain Architecture

In [97]:
seq_with_DomainCount.groupby('Organism').DomainCount.value_counts()

Organism                                                       DomainCount
Aegilops tauschii (Tausch's goatgrass) (Aegilops squarrosa)    1              1
Aegilops tauschii subsp. strangulata (Goatgrass)               1              2
Ananas comosus (Pineapple) (Ananas ananas)                     1              2
Apostasia shenzhenica                                          1              1
Arabidopsis lyrata subsp. lyrata (Lyre-leaved rock-cress)      1              1
                                                                             ..
Triticum aestivum (Wheat)                                      1              3
Triticum turgidum subsp. durum (Durum wheat) (Triticum durum)  1              1
Triticum urartu (Red wild einkorn) (Crithodium urartu)         1              1
Vitis vinifera (Grape)                                         1              1
Zea mays (Maize)                                               1              2
Name: DomainCount, Length: 147, dtype: int64

In [99]:
## seq_with_DomainCount.groupby('Organism').DomainCount.value_counts().to_csv('data/31_Organism_wise_distribution_of_DomainArchitecture.csv')

In [101]:
## seq_with_DomainCount.to_csv('data/32_Pin_II_sequences_with_DOmainCount.csv',index=False)